File: Algorithms/Searching/Binary Search/BinarySearch.java
import java.util.Arrays;

class BinarySearch {
    public static void main(String[] args) {
        int[] data = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 };
        System.out.println("Before: " + Arrays.toString(data));
        int result = sort(data, 8);
        System.out.println("After:  " + result);
    }

    public static int sort(int[] data, int key) {
        int low = 0;
        int high = data.length - 1;
        while(low <= high)
        {
            int mid = low + (high - low) / 2;
            if (key < data[mid]) high = mid - 1;
            else if (key > data[mid]) low = mid + 1;
            else return mid;
        }
        return -1;
    }
}

File: Algorithms/Searching/Binary Search/BinarySearch.py
def binarySearch(arr, low, high, x):
    while low <= high:
        mid = low + (high - low)//2
        if x < arr[mid]:
            high = mid - 1
        elif x > arr[mid]:
            low = mid + 1
        else:
            return mid
    return -1


arr = [1, 5, 4, 8, 12, 2, 3, 22, 7, 9]
print(f"After: {binarySearch(arr, 0, len(arr) - 1, 2)}")